aws/table_aws_sagemaker_training_job.go

```diff
@@ -0,0 +1,430 @@
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "stopping_condition",
+				Description: "Specifies a limit to how long a model training job can run.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "tensor_board_output_config",
+				Description: "Configuration of storage locations for the Debugger TensorBoard output data.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "vpc_config",
+				Description: "A VpcConfig object that specifies the VPC that this training job has access to.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJob,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the training job.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJobTags,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("TrainingJobName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerTrainingJobTags,
+				Transform:   transform.FromField("Tags").Transform(sageMakerTrainingJobTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("TrainingJobArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsSageMakerTrainingJobs(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAwsSageMakerTrainingJobs", "AWS_REGION", region)
+
+	// Create Session
+	svc, err := SageMakerService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.ListTrainingJobsPages(
+		&sagemaker.ListTrainingJobsInput{},
+		func(page *sagemaker.ListTrainingJobsOutput, isLast bool) bool {
+			for _, job := range page.TrainingJobSummaries {
+				d.StreamListItem(ctx, job)
+			}
+			return !isLast
+		},
+	)
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsSageMakerTrainingJob(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var name string
+	if h.Item != nil {
+		name = trainingJobName(h.Item)
```

Comment:
Do we require to check output `type` for `GET` function?
I think the type will be same as `LIST` i.e. `*sagemaker.TrainingJobSummary`